tools/c7n_azure/c7n_azure/resources/web_app.py

```diff
@@ -74,3 +90,19 @@ class resource_type(ArmResourceManager.resource_type):
             'properties.hostNames[0]'
         )
         resource_type = 'Microsoft.Web/sites'
+
+
+@WebApp.filter_registry.register('configuration')
+class ConfigurationFilter(ValueFilter):
+    schema = type_schema('configuration', rinherit=ValueFilter.schema)
+    schema_alias = True
+
+    def __call__(self, i):
+        if 'configuration' not in i:
+            client = self.manager.get_client().web_apps
+            instance = (
+                client.get_configuration(i['resourceGroup'], i['name'])
+            )
+            i['configuration'] = instance.serialize()['properties']
```

Comment:
`    def serialize(self, keep_readonly=False):` -- when we do serialize, we want to keep read only elements available for the filter

Reply:
That makes a lot of sense and is a good practice to follow. I just tested to see if there are any additional fields that are readonly and it looks like there are not any, but who knows how this might change in the future